Source/WebCore/page/Quirks.cpp

```diff
@@ -2160,1043 +2436,988 @@ void Quirks::setTopDocumentURLForTesting(URL&& url)
-    UNUSED_PARAM(documentURL);
     // weather.com rdar://139689157
-    quirksData.needsFormControlToBeMouseFocusableQuirk = true;
+    quirksData.enableQuirk(QuirksData::SiteSpecificQuirk::NeedsFormControlToBeMouseFocusableQuirk);
 }
 #endif
 
-static void handleTikTokQuirks(QuirksData& quirksData, const URL& quirksURL, const String& quirksDomainString, const URL& documentURL)
+static void handleTikTokQuirks(QuirksData& quirksData, const URL& /* quirksURL */, const String& quirksDomainString, const URL&  /* documentURL */)
 {
-    UNUSED_VARIABLE(quirksURL);
-    UNUSED_VARIABLE(documentURL);
-    if (quirksDomainString != "tiktok.com"_s)
+    if (quirksDomainString != "tiktok.com"_s) [[unlikely]]
         return;
 
-    quirksData.needsTikTokOverflowingContentQuirk = true;
+    quirksData.enableQuirk(QuirksData::SiteSpecificQuirk::NeedsTikTokOverflowingContentQuirk);
 }
 
 #if PLATFORM(IOS_FAMILY)
-static void handleDisneyPlusQuirks(QuirksData& quirksData, const URL& quirksURL, const String& quirksDomainString, const URL& documentURL)
+static void handleDisneyPlusQuirks(QuirksData& quirksData, const URL& /* quirksURL */, const String& quirksDomainString, const URL&  /* documentURL */)
 {
-    if (quirksDomainString != "disneyplus.com"_s)
+    if (quirksDomainString != "disneyplus.com"_s) [[unlikely]]
         return;
 
-    UNUSED_PARAM(quirksURL);
-    UNUSED_PARAM(documentURL);
-    // disneyplus rdar://137613110
-    quirksData.shouldHideCoarsePointerCharacteristicsQuirk = true;
+    quirksData.enableQuirks({
+        // disneyplus rdar://137613110
+        QuirksData::SiteSpecificQuirk::ShouldHideCoarsePointerCharacteristicsQuirk,
 #if ENABLE(DESKTOP_CONTENT_MODE_QUIRKS)
-    // disneyplus rdar://151715964
-    quirksData.needsZeroMaxTouchPointsQuirk = true;
+        // disneyplus rdar://151715964
+        QuirksData::SiteSpecificQuirk::NeedsZeroMaxTouchPointsQuirk,
 #endif
+    });
 }
 
-static void handleGuardianQuirks(QuirksData& quirksData, const URL& quirksURL, const String& quirksDomainString, const URL& documentURL)
+static void handleGuardianQuirks(QuirksData& quirksData, const URL& /* quirksURL */, const String& /* quirksDomainString */, const URL&  /* documentURL */)
 {
-    UNUSED_PARAM(quirksURL);
-    UNUSED_PARAM(quirksDomainString);
-    UNUSED_PARAM(documentURL);
-    quirksData.shouldHideSoftTopScrollEdgeEffectDuringFocusQuirk = true;
+    quirksData.enableQuirk(QuirksData::SiteSpecificQuirk::ShouldHideSoftTopScrollEdgeEffectDuringFocusQuirk);
 }
 #endif // PLATFORM(IOS_FAMILY)
 
 #if ENABLE(DESKTOP_CONTENT_MODE_QUIRKS)
-static void handleMaxQuirks(QuirksData& quirksData, const URL& quirksURL, const String& quirksDomainString, const URL& documentURL)
+static void handleMaxQuirks(QuirksData& quirksData, const URL& /* quirksURL */, const String& quirksDomainString, const URL&  /* documentURL */)
 {
-    if (quirksDomainString != "max.com"_s)
+    if (quirksDomainString != "max.com"_s) [[unlikely]]
```

Comment:
We use quite a few checks of `[[unlikely]``. Shouldn't these be ``likely``, since there is a good chance it is not the domain?